pints/tests/test_mcmc_nuts.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python3
+#
+# Tests the basic methods of the Hamiltonian MCMC routine.
+#
+# This file is part of PINTS (https://github.com/pints-team/pints/) which is
+# released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+import numpy as np
+import unittest
+import sys
+
+import pints
+import pints.toy
+
+from shared import StreamCapture
+
+
+@unittest.skipIf(sys.hexversion < 0x03030000, 'No NUTS on Python < 3.3')
+class TestNutsMCMC(unittest.TestCase):
+    """
+    Tests the basic methods of the No-U-Turn MCMC sampler.
+    """
+
+    def test_method(self):
+
+        # Create log pdf
+        log_pdf = pints.toy.GaussianLogPDF([5, 5], [[4, 1], [1, 3]])
+
+        # Create mcmc
+        x0 = np.array([2, 2])
+        sigma = [[3, 0], [0, 3]]
+        mcmc = pints.NoUTurnMCMC(x0, sigma)
+
+        # This method needs sensitivities
+        self.assertTrue(mcmc.needs_sensitivities())
+
+        # Perform short run
+        chain = []
+        for i in range(2 * mcmc.number_adaption_steps()):
+            x = mcmc.ask()
+            fx, gr = log_pdf.evaluateS1(x)
+            sample = mcmc.tell((fx, gr))
+            if sample is not None:
+                chain.append(sample)
+            if np.all(sample == x):
+                self.assertEqual(mcmc.current_log_pdf(), fx)
+
+        chain = np.array(chain)
+        self.assertGreater(chain.shape[0], 1)
+        self.assertEqual(chain.shape[1], len(x0))
+
+    def test_logging(self):
+        """
```

Comment:
Unfortunately we can't have docstrings inside `test_x` methods because `unittest` displays them instead of the test name (making the actual test harder to find), so please use `# Test logging includes...`